### DISCUSSION TITLE
Connecting to the Solar System

### DISCUSSION THREAD

**Discussion by man1ac85** [wlcrs/huawei_solar#680] [Setting up the integration]
Hey guys,

we recently installed a solar system in Portugal at our house.  For various reasons we bought a fully configured system from EDP (they install and do all the regulartory stuff we cant handle).

I now want to install a Home Assitant System to use the solar power most efficiently. We got various loads we want to switch on/off depending on the solar power available.
I read through the whole article on how to connect to the inverter. We identified the COM Ports. But I think we have a real issue: We dont have a admin/installateur account on fusion solar. Only the the "standard user" stuff which lets us see all the data.

I tried to find the settings to setup but the communication config but there are no settings like this anywhere.

Any recommendation on what I can do or test? There is the 4G Dongle installed (if that matters for any reason).

Thanks and regards
Bastian

**Comment by andcompe** [comment]
Hi, 

I also had some issues with connecting througth the serial port (that is another story) and I cannot use a wifi dongle in my HA system since it is too far away. So I came up with a small solution that has actually worked quite well since several months back. 

If you are familiar with Arduino or ESP8266, it will be easy to use. I have put it out on github recently. This solution only require that you can connect to the inverter using the default AP. 

https://github.com/andcompe/HuaweiSolarBridge/tree/main

Perhaps it could be an alternative for you.

**Reply by man1ac85** [reply]
Thanks! Ill have a read into that and get back to you :)
Programming some ESP wont be an issue, I hope :D

**Comment by Roving-Ronin** [comment]
@man1ac85 
You could also either see about logging in locally with the SolarFusion app (mobile phone) and resetting the setup to factory default, create your own installer account and then setup the inverter under your account. You'd then have to grant access to EDP if you want them to still be able to have any access for them (usually a pain in the ####, as most installers don't see to proactively update firmware as needed with fixes coming out... something that needs an installer account to do).

Otherwise, assuming your Portugal place has an permanent internet connection and WiFi operating, you could see about replacing the 4G dongle with the sDongle-A5 (WiFi / Fast Ethernet), then asking EDP to enable modbus-tcp access upon it from the local network (and assign the sDongle a static/reserved IP on your home router/DHCP server).

This would then see the setup continue to be able to communicate with the FusionSolar portal (i.e. intl.fusionsolar.huawei.com ), which is where EDP would be logging into to access any data upon it / change any settings. Whilst given its then on your home WiFi would also mean it is now available for you to query via modbus-tcp.

Depending upon the inverter model, they actually some with an sDongle included (i.e. L1 and M1 models I know do).
